### PR TITLE
WIP: django_stubs_ext: monkeypatch `reveal_{type,locals}` into builtins

### LINKED ISSUE #590: 'reveal_type' is not defined when using mypy_django_plugin
Using `reveal_type` in a module fails when using `plugins = project.mypy_django_plugin` (via `[mypy]` section in `setup.cfg`):

```
% mypy project/app/admin.py
Error constructing plugin instance of NewSemanalDjangoPlugin

Traceback (most recent call last):
  File "…/project/.venv/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "…/project/.venv/lib/python3.9/site-packages/mypy/__main__.py", line 11, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "mypy/main.py", line 90, in main
  File "mypy/build.py", line 179, in build
  File "mypy/build.py", line 228, in _build
  File "mypy/build.py", line 472, in load_plugins
  File "mypy/build.py", line 450, in load_plugins_from_config
  File "…/project/.venv/lib/python3.9/site-packages/mypy_django_plugin/main.py", line 104, in __init__
    self.django_context = DjangoContext(django_settings_module)
  File "…/project/.venv/lib/python3.9/site-packages/mypy_django_plugin/django/context.py", line 88, in __init__
    apps, settings = initialize_django(self.django_settings_module)
  File "…/project/.venv/lib/python3.9/site-packages/mypy_django_plugin/django/context.py", line 72, in initialize_django
    apps.populate(settings.INSTALLED_APPS)
  File "…/project/.venv/lib/python3.9/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "…/project/.venv/lib/python3.9/site-packages/django/contrib/admin/apps.py", line 24, in ready
    self.module.autodiscover()
  File "…/project/.venv/lib/python3.9/site-packages/django/contrib/admin/__init__.py", line 24, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "…/project/.venv/lib/python3.9/site-packages/django/utils/module_loading.py", line 47, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "…/project/app/operations/admin.py", line 928, in <module>
    reveal_type(FooAdmin)
NameError: name 'reveal_type' is not defined
```

## System information

- OS:
- `python` version: Python 3.9.2
- `django` version: Django 3.1.8
- `mypy` version: mypy 0.812
- `django-stubs` version: f003968 (current master)

### PR DESCRIPTION
**Pull request by blueyed**
Closes https://github.com/typeddjango/django-stubs/issues/590.

### CODE REVIEW

**Comment by blueyed** [comment]
Any suggestions for a test, and/or how to proceed here in general?

File: django_stubs_ext/django_stubs_ext/patch.py [hunk 1]
```diff
@@ -46,15 +46,24 @@ def __repr__(self) -> str:
...
     suited_for_this_version = filter(
         lambda spec: spec.version is None or VERSION[:2] <= spec.version,
         _need_generic,
     )
     for el in suited_for_this_version:
         el.cls.__class_getitem__ = classmethod(lambda cls, *args, **kwargs: cls)
 
+    # Define reveal_type/reveal_locals, to not cause NameError during setting
+    # up Django.
+    if not TYPE_CHECKING:
```
Comment (sobolevn) [inline django_stubs_ext/django_stubs_ext/patch.py]: We assume that all code under `monkeypatch` is called during runtime and not type-checking.

File: django_stubs_ext/django_stubs_ext/patch.py [hunk 2]
```diff
@@ -46,15 +46,24 @@ def __repr__(self) -> str:
...
+        import builtins
```
Comment (sobolevn) [inline django_stubs_ext/django_stubs_ext/patch.py]: So, we can move this to the module level.

**Review by sobolevn — COMMENTED** [review]
Thanks!

**Comment by sobolevn** [comment]
> Any suggestions for a test, and/or how to proceed here in general?

We can test that after calling `monkeypatch`, we can use `reveal_type` in our regular python code.

**Comment by blueyed** [comment]
Added some rough test for now - happy to improve it later.

**Review by sobolevn — APPROVED** [review]
Awesome! 👍

**Comment by sobolevn** [comment]
Looks like I need to release `django_stubs_ext`. Will do later today!

**Comment by blueyed** [comment]
> Looks like I need to release `django_stubs_ext`. Will do later today!

Let's hold this back a bit: https://github.com/typeddjango/django-stubs/issues/590#issuecomment-819423678

**Pull request merged**